pkg/client_data/lib/account_api.dart

```diff
@@ -2,11 +2,51 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:convert';
+
 import 'package:json_annotation/json_annotation.dart';
 import 'package:meta/meta.dart';
 
 part 'account_api.g.dart';
 
+/// Session data to be cached from the authenticated client.
+@JsonSerializable()
+class ClientSessionData {
+  final String imageUrl;
+
+  ClientSessionData({
+    @required this.imageUrl,
```

Comment:
We could also upload the `access_token` and then use it to call the userinfo end-point on the remote side..

We really ought to verify that:
 A) the `access_token` works with the `tokeninfo` end-point,
 B) It has one our allowed `audience`s,
 C) It has the right `user_id`.
We should do (A), (B) and (C) before calling the userinfo end-point.

Finally, we should verify (A), (B) and (C) when we do `createPublisher`.

I think `AccountBackend` has a method to get the user from an `access_token`, we just need to use that..

-------
Sorry for the scope creep, we can do `createPublisher` changes in a different PR.

Reply:
This seems to be a larger change, let's do it separately.

Reply:
(especially as changing the access_token to id_token in the auth headers seems to need additional work)

Reply:
I'm okay doing this separately, can we file an issue and the land this.